Automattic-Tracks-iOS/Services/EventLoggingUploadQueue.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+
+open class EventLoggingUploadQueue {
+
+    private let fileManager = FileManager.default
+
+    /// The log file on top of the queue
+    var first: LogFile? {
+
+        guard let url = try? fileManager.contentsOfDirectory(at: storageDirectory).first else {
+            return nil
+        }
+
+        return LogFile(url: url, uuid: url.lastPathComponent)
+    }
+
+    func add(_ log: LogFile) throws {
+        try ensureStorageDirectoryExists()
+        try fileManager.copyItem(at: log.url, to: storageURL(forLog: log))
+    }
+
+    func remove(_ log: LogFile) throws {
+        let url = storageURL(forLog: log)
+        if fileManager.fileExistsAtURL(url) {
+            try fileManager.removeItem(at: storageURL(forLog: log))
+        }
+    }
+
+    var storageDirectory: URL {
```

Comment:
Question: Is there a convention to keep the properties at the top of the file in iOS? That's what we do in Android, but it might be different here.

Reply:
Good question – for "normal" properties, I agree that this makes sense.

For the case of calculated properties like this, because they could be represented as `func storageDirectory() -> URL`, IMHO it's essentially a method, and so I like not cluttering up the top of my class with it, but I'm not aware of a code style rule for this.